sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -604,6 +604,25 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("bucket join should work with SubqueryAlias plan") {
+    withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "0") {
+      withTable("t") {
+        withView("v") {
+          val df = (0 until 20).map(i => (i, i)).toDF("i", "j").as("df")
```

Comment:
nit: We need `.as("df")` for this test? I think you can just write it like;
```
spark.range(20).selectExpr("id as i", "id as j").write.bucketBy(8, "I").saveAsTable("t")
```

Reply:
Thanks, updated.